src/test/java/core/basesyntax/operation/OperationStrategyTest.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.model.Record;
+import core.basesyntax.report.FruitBalance;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class OperationStrategyTest {
+    private static final String BALANCE = "b";
+    private static final String SUPPLY = "s";
+    private static final String PURCHASE = "p";
+    private static final String RETURN = "r";
+    private static final String EQUAL_MAP_KEY = "apple";
+    private static final int AMOUNT_ADDITION = 20;
+    private static final int AMOUNT_DECREASE = 5;
+    private static Map<String, OperationHandler> operationHandlerMap;
+    private static OperationStrategy operationStrategy;
+
+    @BeforeClass
+    public static void beforeAll() {
+        operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(BALANCE, new AdditionHandler());
+        operationHandlerMap.put(SUPPLY, new AdditionHandler());
+        operationHandlerMap.put(PURCHASE, new DecreaseHandler());
+        operationHandlerMap.put(RETURN, new AdditionHandler());
+        operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+    }
+
+    @Test
+    public void get_balanceOperation_Ok() {
+        operationHandlerMap.get(BALANCE).apply(new Record(BALANCE, "apple", 10));
+        operationStrategy.get(BALANCE).apply(new Record(BALANCE, "apple", 10));
+        int actual = FruitBalance.FRUIT_BALANCE.get(EQUAL_MAP_KEY);
+        int expected = AMOUNT_ADDITION;
```

Comment:
What class are you testing here: **_OperationStrategyImpl_** or one of your handlers classes?

Reply:
Reworked